core/accounting/src/accounting_period/mod.rs

```diff
@@ -0,0 +1,305 @@
+        let open_periods = self.repo.find_open_accounting_periods(chart_id).await?;
+
+        let chart = self.chart_of_accounts.find_by_id(chart_id).await?;
+
+        let chart_config = self
+            .ledger
+            .get_chart_of_accounts_integration_config(tracking_account_set_id)
+            .await?
+            .ok_or(AccountingPeriodError::AccountingPeriodIntegrationConfigNotFound)?;
+
+        let account_set_ids = AccountingPeriodAccountSetIds {
+            tracking_account_set_id,
+            revenue_account_set_id: chart.account_set_id_from_code(&chart_config.revenue_code)?,
+            cost_of_revenue_account_set_id: chart
+                .account_set_id_from_code(&chart_config.cost_of_revenue_code)?,
+            expenses_account_set_id: chart.account_set_id_from_code(&chart_config.expenses_code)?,
+            equity_retained_earnings_account_set_id: chart
+                .account_set_id_from_code(&chart_config.equity_retained_earnings_code)?,
+            equity_retained_losses_account_set_id: chart
+                .account_set_id_from_code(&chart_config.equity_retained_losses_code)?,
+        };
+
+        if open_periods.is_empty() {
+            for period in periods {
+                let period = match period.basis {
+                    Basis::Monthly { day } => {
+                        Period::monthly_around_date(day, date, period.grace_period_days).unwrap()
+                    }
+                    Basis::Annual { day, month } => {
+                        Period::annually_around_date(day, month, date, period.grace_period_days)
+                            .unwrap()
+                    }
+                };
+
+                self.repo
+                    .create(NewAccountingPeriod {
+                        id: AccountingPeriodId::new(),
+                        chart_id,
+                        period,
+                        account_set_ids,
+                        closed_at: None,
+                    })
+                    .await?;
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Closes currently open monthly Accounting Period under the given
+    /// Chart of Accounts and returns next Accounting Period.
+    /// Fails if no such Accounting Period is found.
+    pub async fn close_month(
+        &self,
+        sub: &<<Perms as PermissionCheck>::Audit as AuditSvc>::Subject,
+        chart_id: ChartId,
+    ) -> Result<AccountingPeriod, AccountingPeriodError> {
+        self.authz
+            .enforce_permission(
+                sub,
+                CoreAccountingObject::all_accounting_period(),
+                CoreAccountingAction::ACCOUNTING_PERIOD_CLOSE,
+            )
+            .await?;
+        let mut open_periods = self.repo.find_open_accounting_periods(chart_id).await?;
+
+        let open_period = open_periods
+            .iter_mut()
+            .find(|p| p.is_monthly())
+            .ok_or(AccountingPeriodError::NoOpenAccountingPeriodFound)?;
```

Comment:
Nit, can optionally assign directly to `open_periods` (for readability), and introduce more granular error (`NoOpenMonthlyAccountingPeriodFound`) to help with debugging later on

```suggestion
        let mut open_period = self
            .repo
            .find_open_accounting_periods(chart_id)
            .await?
            .into_iter()
            .find(|p| p.is_monthly())
            .ok_or(AccountingPeriodError::NoOpenMonthlyAccountingPeriodFound)?;
```